modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1395,7 +1400,8 @@
   "Compile an MBQL query."
   [query]
   (let [query (update query :query preprocess)]
-    (binding [*query* query]
+    (binding [*query* query
+              *next-alias-index* (atom 0)]
```

Comment:
_Why use binding to atom?_ My current mental model is that we could have multiple threads compiling multiple mongo queries at once. As biding creates per-thread binding, this ensures that every compilation has its own counter.

I've decided to use binding as I think sending metadata (how many joins were compiled) as return values and arguments of compilation functions would make code more convoluted. If this approach is not desired, I'll change it.

Reply:
We could emphasize the fact that this counter is not to be shared by using `volatile!` instead of `atom`.

Reply:
Done. I'll mind that for future. Thanks.